### PR TITLE
Add org.kde.kontact

### PR DESCRIPTION
**Pull request by danvratil**

### CODE REVIEW

File: org.kde.kontact.json [hunk 1]
```diff
@@ -0,0 +1,1233 @@
...
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "kdiagram",
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
+                "-DCMAKE_INSTALL_LIBDIR=lib"
```
Comment (eszlari) [inline org.kde.kontact.json]: This is not needed anymore.

File: org.kde.kontact.json [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://dl.bintray.com/boostorg/release/1.68.0/source/boost_1_68_0.tar.bz2",
+                    "sha256": "7f6130bc3cf65f56a618888ce9d5ea704fa10b462be126ad053e80e553d6d8b7"
+                },
+                {
+                    "type": "file",
+                    "path": "boost-configure",
```
Comment (eszlari) [inline org.kde.kontact.json]: By using `buildsystem: simple` you could get ride of this file, see e.g.:
https://github.com/flathub/org.kde.krita/blob/master/org.kde.krita.yaml#L29

**Comment by eszlari** [comment]
You might want to add `-DCMAKE_BUILD_TYPE=RelWithDebInfo` to all modules that are using cmake.

File: org.kde.kontact.json [hunk 3]
```diff
@@ -0,0 +1,1233 @@
...
+                    "cleanup": [
+                        "/share/gtk-doc"
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "boost",
+            "config-opts": [
+                "--with-libraries=graph,program_options,system,thread"
```
Comment (eszlari) [inline org.kde.kontact.json]: You can drop that now.

File: org.kde.kontact.json [hunk 4]
```diff
@@ -0,0 +1,1233 @@
...
+                }
+            ],
+            "modules": [
+                {
+                    "name": "sqlite3",
+                    "config-opts": [
+                        "--enable-threadsafe",
+                        "--enable-threads-override-locks"
+                    ],
+                    "cmake": false,
```
Comment (eszlari) [inline org.kde.kontact.json]: This can be dropped.

File: org.kde.kontact.json [hunk 5]
```diff
@@ -0,0 +1,1233 @@
...
+                    "url": "https://download.kde.org/stable/applications/18.08.1/src/kdepim-runtime-18.08.1.tar.xz",
+                    "sha256": "92cf5dad62236af2b7a7eddd19e11c9b7d41de45574c59c2fbefb9f00d6a6304"
+                }
+            ],
+            "modules": [
+                {
+                    "name": "libkolabxml",
+                    "config-opts": [
+                        "-DBUILD_TESTS=OFF",
+                        "-DQT5_BUILD=ON"
```
Comment (eszlari) [inline org.kde.kontact.json]: `-DCMAKE_BUILD_TYPE=RelWithDebInfo`

**Comment by eszlari** [comment]
Maybe you should mention that you are upstream and tell the flathub admin who else should get commit access.

**Comment by nedrichards** [comment]
bot, build org.kde.kontact

**Comment by nedrichards** [comment]
Builds failed on ARM and aarch64: https://flathub.org/builds/#/builders/2/builds/4723 will try and provide some more detailed advice later.

File: org.kde.kontact.json [hunk 6]
```diff
@@ -0,0 +1,1224 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/frameworks/5.50/prison-5.50.0.tar.xz",
+                    "sha256": "4d98254c787f4ec4f80317a738d4f5ec6f2125144df1185fe61c444a086ec2ec"
+                }
+            ],
+            "modules": [
+                {
+                    "name": "qrencode",
```
Comment (nedrichards) [inline org.kde.kontact.json]: This needs to be fixed for aarch64 exactly like this: https://github.com/flathub/cx.ring.Ring/commit/b024093f5115e874a567a5c7d049fbc70efea839#diff-2e592000f12516bca751f85f029e3ee3R169

File: org.kde.kontact.json [hunk 7]
```diff
@@ -0,0 +1,1224 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/applications/18.08.1/src/kmail-18.08.1.tar.xz",
+                    "sha256": "747eac028620da6c6aaf9b00d0d6ab3eabea23dcbad294b26ebff4c6ac3c0988"
+                }
+            ]
+        },
+        {
+            "name": "akregator",
```
Comment (nedrichards) [inline org.kde.kontact.json]: This build appears to fail on ARM, nothing terribly obvious why - https://flathub.org/builds/#/builders/48/builds/415

File: org.kde.kontact.json (hunk 6 above)
Comment (eszlari) [reply org.kde.kontact.json]: Or you could try cmake.

File: org.kde.kontact.json (hunk 7 above)
Comment (danvratil) [reply org.kde.kontact.json]: Looks like a bug in Qt moc - the generated code refers to a class in a wrong namespace. I added a patch that tries to solve it by moving the class declaration to the namespace where moc expects it, let's see if it helps :)

File: org.kde.kontact.json (hunk 6 above)
Comment (nedrichards) [reply org.kde.kontact.json]: that would actually likely be better!

**Comment by nedrichards** [comment]
bot, build org.kde.kontact

File: org.kde.kontact.json (hunk 6 above)
Comment (danvratil) [reply org.kde.kontact.json]: Oki, switched to cmake in qrencode. Let's hope that will do it :)

**Comment by danvratil** [comment]
Looks like LDAP does not like aarch64 either :/ I tried to apply the same fix as you suggested for qrencode.

Sorry for the trial&error approach, I can't test this very well locally :)

**Comment by nedrichards** [comment]
no worries - co-incidentally Evolution just fixed openldap for aarch64 as well - it's a race! https://github.com/flathub/flathub/pull/616/files ;-)

**Comment by nedrichards** [comment]
bot, build org.kde.kontact

**Comment by barthalion** [comment]
bot, build org.kde.kontact

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.kde.kontact

**Comment by nedrichards** [comment]
@danvratil tell us the github usernames of anyone else from upstream who needs push access to the newly created repo

**Comment by danvratil** [comment]
Many thanks, guys!

Could you please add @Montel and @hefee to the repo?

**Comment by barthalion** [comment]
Added; make sure to accept invitations to get write access.

**Pull request closed without merging**